src/view.coffee

```diff
@@ -1642,12 +1646,46 @@ define ['droplet-helper', 'droplet-draw', 'droplet-model'], (helper, draw, model
 
         # Blocks have a shape including a lego nubby "tab", and so
         # they need to be at least wide enough for tabWidth+tabOffset.
+
+        @extraWidth = 0
+        if 'add-button' in @model.classes
+          @extraWidth += @view.opts.buttonWidth + @view.opts.buttonPadding
+
+        if 'subtract-button' in @model.classes
+          @extraWidth += @view.opts.buttonWidth + @view.opts.buttonPadding
+
         for size, i in @minDimensions
           size.width = Math.max size.width,
               @view.opts.tabWidth + @view.opts.tabOffset
+          size.width += @extraWidth
 
         return null
 
+      drawSelf: (ctx, style) ->
+        super
+
+        start = @totalBounds.x + @totalBounds.width - @extraWidth
```

Comment:
Need to think through positioning in the multiline case.  Currently you put the +/- buttons in the upper-right corner of the bounding rect for the block.  We want it to be on the last line.  But lower-right corner isn't right: you should figure the actual width of the last line to position the +/-.